lib/vdc/ArbitrarilyOrientedRegularGrid.cpp

```diff
@@ -0,0 +1,228 @@
+#include <limits>
+#include <iomanip>
+
+#include <vapor/utils.h>
+
+#include <glm/gtc/type_ptr.hpp>
+#include <GTE/MinimumAreaBox2.h>
+#include <vapor/RegularGrid.h>
+#include <vapor/ArbitrarilyOrientedRegularGrid.h>
+
+using namespace std;
+using namespace VAPoR;
+
+// clang-format off
+
+ArbitrarilyOrientedRegularGrid::ArbitrarilyOrientedRegularGrid(
+    const VAPoR::Grid *grid3d,
+    planeDescription& pd,
+    const DimsType& dims,
+    std::shared_ptr<float>& blks
+) : RegularGrid(
+        dims, 
+        {{pd.sideSize, pd.sideSize, 1}}, 
+        {(_myBlks = new float[dims[0]*dims[1]*dims[2]])}, 
+        {{0.,0.,0.}}, 
+        {{1.,1.,1.}}
+) {
+    SetMissingValue( grid3d->GetMissingValue() );
+
+    _sideSize = pd.sideSize;
+    _rotation = {pd.rotation[0], pd.rotation[1], pd.rotation[2]};
+
+    CoordType gridMin, gridMax;
+    grid3d->GetUserExtents(gridMin, gridMax);
+
+    // .vs3 files before Vapor3.6 will try to initialize Slices with origin values set to 0, which
+    // can sometimes lie outside of the domain.  If this happens, we need to configure a new origin
+    // within the 3D grid's bounds.
+    for (int i=0; i<3; i++) {
+        _origin[i] = (pd.origin[i] <= gridMin[i] && pd.origin[i] >= gridMax[i]) ? pd.origin[i] : (gridMax[i]-gridMin[i])/2 + gridMin[i];
+        if (pd.origin[i] <= gridMax[i] && pd.origin[i] >= gridMin[i]) 
+            _origin[i] = pd.origin[i];
+        else 
+            _origin[i] = (gridMax[i]-gridMin[i])/2 + gridMin[i];
+    }
+
+    // Rotate the plane via quaternion method
+    _rotate();
+
+    // Find the vertices where our plane intercepts the edges of the Box
+    std::vector<glm::tvec3<double, glm::highp>> vertices;
+    _findIntercepts(gridMin, gridMax, vertices);
+
+    // Find the minimum-area-rectangle that encloses the vertices that are along the edges
+    // of the Box.  Define this rectangle in 3D space, and in a newly projecteed 2D space.
+    std::vector<glm::tvec3<double, glm::highp>> tmpRectangle3D;
+    _getMinimumAreaRectangle( vertices );
+
+    // Pick sample points along our 2D rectangle, and project those points back into 3D space
+    // to query our 3D grid for data values.
+    _populateData( grid3d, pd, blks );
```

Comment:
This should be `_myBlks`, not `blks` shouldn't it?

Reply:
Yep.  This change was on a separate branch but not this one.  Updated and thanks for the catch.